In [1]:
import basedosdados as bd
import pandas as pd

query = "SELECT * FROM `datario.adm_central_atendimento_1746.chamado`" \
        "WHERE DATE(data_inicio) = '2023-04-01'"
df_chamado = bd.read_sql(query, billing_project_id="prefeitura-rio-455211")
df_chamado.head()

Downloading: 100%|██████████|


id_chamado id_origem_ocorrencia         data_inicio            data_fim  \
0   18517973                   13 2023-04-01 18:03:14 2023-05-02 15:22:24   
1   18518136                   13 2023-04-01 19:15:24 2023-04-03 16:31:06   
2   18516836                    1 2023-04-01 10:34:10 2023-04-03 14:07:40   
3   18516688                    1 2023-04-01 09:53:57 2023-04-27 16:31:54   
4   18517231                   13 2023-04-01 12:45:19 2023-05-02 09:21:23   

  id_bairro id_territorialidade id_logradouro  numero_logradouro  \
0        81                   3         25544                974   
1      None                None          None               <NA>   
2      None                None          None               <NA>   
3      None                None          None               <NA>   
4      None                None          None               <NA>   

  id_unidade_organizacional nome_unidade_organizacional  ...  \
0                       247           Ouvidoria COMLURB  ...   
1                       121            Ouvidoria GM-RIO  ...   
2                       835        Ouvidoria SMS/CAP3.1  ...   
3                        43              Ouvidoria SMTR  ...   
4                       846                OUV.SMS/HMSA  ...   

  data_real_diagnostico tempo_prazo prazo_unidade prazo_tipo   dentro_prazo  \
0                   NaT        <NA>             D          F  Fora do prazo   
1                   NaT        <NA>             D          F       No prazo   
2                   NaT        <NA>             D          F       No prazo   
3                   NaT        <NA>             D          F       No prazo   
4                   NaT        <NA>             D          F  Fora do prazo   

    situacao          tipo_situacao justificativa_status  reclamacoes  \
0  Encerrado               Atendido                 None            0   
1  Encerrado  Atendido parcialmente                 None            0   
2  Encerrado  Atendido parcialmente                 None            0   
3  Encerrado               Atendido                 None            0   
4  Encerrado  Atendido parcialmente                 None            0   

   data_particao  
0     2023-04-01  
1     2023-04-01  
2     2023-04-01  
3     2023-04-01  
4     2023-04-01  

[5 rows x 32 columns]

In [10]:
query_bairro = "SELECT * FROM `datario.dados_mestres.bairro`"
df_bairro = bd.read_sql(query_bairro, billing_project_id="prefeitura-rio-455211")
df_bairro.head()

Downloading: 100%|██████████|


id_bairro          nome id_area_planejamento id_regiao_planejamento  \
0         2        Gamboa                    1                    1.1   
1         1         Saúde                    1                    1.1   
2         4          Caju                    1                    1.1   
3         3  Santo Cristo                    1                    1.1   
4       161          Lapa                    1                    1.1   

  nome_regiao_planejamento id_regiao_administrativa  \
0                   Centro                        1   
1                   Centro                        1   
2                   Centro                        1   
3                   Centro                        1   
4                   Centro                        2   

  nome_regiao_administrativa subprefeitura          area     perimetro  \
0                  Portuaria        Centro  1.112903e+06   4612.833630   
1                  Portuaria        Centro  3.638176e+05   2646.220568   
2                  Portuaria        Centro  5.347481e+06  19800.522524   
3                  Portuaria        Centro  1.684721e+06   6743.227885   
4                     Centro        Centro  2.983258e+05   3849.181818   

                                        geometry_wkt  \
0  POLYGON ((-43.18791509600138 -22.8931217212322...   
1  POLYGON ((-43.181151633502964 -22.895430284304...   
2  MULTIPOLYGON (((-43.22522241788469 -22.8746498...   
3  POLYGON ((-43.194498082949806 -22.903378003392...   
4  POLYGON ((-43.18166120770202 -22.9120798224676...   

                                            geometry  
0  POLYGON((-43.1879150960014 -22.8931217212322, ...  
1  POLYGON((-43.181151633503 -22.8954302843042, -...  
2  MULTIPOLYGON(((-43.196711909178 -22.8872495021...  
3  POLYGON((-43.1944980829498 -22.9033780033923, ...  
4  POLYGON((-43.181661207702 -22.9120798224677, -...

In [3]:
df_chamado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1903 entries, 0 to 1902
Data columns (total 32 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   id_chamado                        1903 non-null   object        
 1   id_origem_ocorrencia              1903 non-null   object        
 2   data_inicio                       1903 non-null   datetime64[us]
 3   data_fim                          1886 non-null   datetime64[us]
 4   id_bairro                         1772 non-null   object        
 5   id_territorialidade               1772 non-null   object        
 6   id_logradouro                     1772 non-null   object        
 7   numero_logradouro                 1649 non-null   Int64         
 8   id_unidade_organizacional         1903 non-null   object        
 9   nome_unidade_organizacional       1903 non-null   object        
 10  id_unidade_organizacional_mae     1903 non-null 

In [11]:
df_bairro.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 164 entries, 0 to 163
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   id_bairro                   164 non-null    object 
 1   nome                        164 non-null    object 
 2   id_area_planejamento        164 non-null    object 
 3   id_regiao_planejamento      164 non-null    object 
 4   nome_regiao_planejamento    164 non-null    object 
 5   id_regiao_administrativa    164 non-null    object 
 6   nome_regiao_administrativa  164 non-null    object 
 7   subprefeitura               164 non-null    object 
 8   area                        164 non-null    float64
 9   perimetro                   164 non-null    float64
 10  geometry_wkt                164 non-null    object 
 11  geometry                    164 non-null    object 
dtypes: float64(2), object(10)
memory usage: 15.5+ KB


## Localização de chamados do 1746


### Q1) Quantos chamados foram abertos no dia 01/04/2023?

In [ ]:
print("Total de chamados em 01/04/2023:", len(df_chamado))

Total de chamados em 01/04/2023: 1903


### Q2) Qual o tipo de chamado que teve mais teve chamados abertos no dia 01/04/2023?


In [9]:
tipo_mais_comum = df_chamado['tipo'].value_counts().idxmax()
contagem_tipo_comum = len(df_chamado[df_chamado['tipo'] == tipo_mais_comum])

print(f"Tipo com mais chamados: {tipo_mais_comum} - {contagem_tipo_comum} chamados")

Tipo com mais chamados: Estacionamento irregular - 373 chamados


### Q3) Quais os nomes dos 3 bairros que mais tiveram chamados abertos nesse dia?

In [15]:
df_chamados_bairro = df_chamado.merge(df_bairro, on='id_bairro', how='left')

bairros_top3 = df_chamados_bairro['nome'].value_counts().head(3)
print("Top 3 bairros com mais chamados:")
print(bairros_top3)

Top 3 bairros com mais chamados:
nome
Campo Grande       124
Tijuca              96
Barra da Tijuca     60
Name: count, dtype: int64


### Q4) Qual o nome da subprefeitura com mais chamados abertos nesse dia?

In [18]:
subprefeitura_top1 = df_chamados_bairro['subprefeitura'].value_counts().head(1)
print("Subprefeitura com mais chamados:")
print(subprefeitura_top1)

Subprefeitura com mais chamados:
subprefeitura
Zona Norte    534
Name: count, dtype: int64


### Q5) Existe algum chamado aberto nesse dia que não foi associado a um bairro ou subprefeitura na tabela de bairros? Se sim, por que isso acontece?

In [24]:
# Caso o id_bairro (identificador) não seja preenchido na ocorrência do chamado, não é possível identificar o bairro ou subprefeitura associado.

sem_bairro_ou_sub = df_chamado[df_chamado["id_bairro"].isnull()]

print("Chamados em 01/04/2023 sem bairro ou subprefeitura:", len(sem_bairro_ou_sub))

Chamados em 01/04/2023 sem bairro ou subprefeitura: 131


## Chamados do 1746 em grandes eventos

In [27]:
query_perturbacao = """
                    SELECT *
                    FROM `datario.adm_central_atendimento_1746.chamado`
                    WHERE id_subtipo = '5071'
                    AND DATE(data_inicio) BETWEEN '2022-01-01' AND '2024-12-31'        
                    """
df_perturbacao = bd.read_sql(query_perturbacao, billing_project_id="prefeitura-rio-455211")
df_perturbacao.info()

Downloading: 100%|██████████|
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56785 entries, 0 to 56784
Data columns (total 32 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   id_chamado                        56785 non-null  object        
 1   id_origem_ocorrencia              56785 non-null  object        
 2   data_inicio                       56785 non-null  datetime64[us]
 3   data_fim                          55761 non-null  datetime64[us]
 4   id_bairro                         55836 non-null  object        
 5   id_territorialidade               55836 non-null  object        
 6   id_logradouro                     55836 non-null  object        
 7   numero_logradouro                 53571 non-null  Int64         
 8   id_unidade_organizacional         56785 non-null  object        
 9   nome_unidade_organizacional       56785 non-null  object        
 10  id_unidade_organ

In [28]:
query_eventos = "SELECT * FROM `datario.turismo_fluxo_visitantes.rede_hoteleira_ocupacao_eventos`"
df_evento = bd.read_sql(query_eventos, billing_project_id="prefeitura-rio-455211")
df_evento.info()

Downloading: 100%|██████████|
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19 entries, 0 to 18
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   ano            19 non-null     object 
 1   data_inicial   6 non-null      dbdate 
 2   data_final     6 non-null      dbdate 
 3   evento         19 non-null     object 
 4   taxa_ocupacao  4 non-null      float64
dtypes: dbdate(2), float64(1), object(2)
memory usage: 892.0+ bytes


### Q6) Quantos chamados de Perturbação do sossego foram abertos nesse período?

In [29]:
print(f"Quantidade de chamados de perturbação do sossego no período: {len(df_perturbacao)}")

Quantidade de chamados de perturbação do sossego no período: 56785


### Q7) Selecione os chamados com esse subtipo que foram abertos durante os eventos contidos na tabela de eventos (Reveillon, Carnaval e Rock in Rio).

In [ ]:
# Adicionando eventos faltantes na tabela
eventos_manualmente = pd.DataFrame([
    {"evento": "Rock in Rio", "data_inicial": pd.to_datetime("2024-09-13"), "data_final": pd.to_datetime("2024-09-15")},
    {"evento": "Rock in Rio", "data_inicial": pd.to_datetime("2024-09-19"), "data_final": pd.to_datetime("2024-09-22")},
    {"evento": "Réveillon",   "data_inicial": pd.to_datetime("2024-12-29"), "data_final": pd.to_datetime("2025-01-01")},
])

eventos_filtrados = df_evento[
    df_evento['evento'].isin(['Carnaval', 'Rock in Rio', 'Réveillon']) &
    df_evento['data_inicial'].notnull() &
    df_evento['data_final'].notnull()
]

df_eventos_completos = pd.concat([eventos_filtrados, eventos_manualmente], ignore_index=True)
df_eventos_completos['data_inicial'] = pd.to_datetime(df_eventos_completos['data_inicial']).dt.date
df_eventos_completos['data_final'] = pd.to_datetime(df_eventos_completos['data_final']).dt.date

In [47]:
df_perturbacao['data_inicio'] = pd.to_datetime(df_perturbacao['data_inicio']).dt.date

df_perturbacao['evento'] = None 

for _, row in df_eventos_completos.iterrows():
    mask = (df_perturbacao['data_inicio'] >= row['data_inicial']) & \
           (df_perturbacao['data_inicio'] <= row['data_final']) & \
           (df_perturbacao['evento'].isnull()) 
    df_perturbacao.loc[mask, 'evento'] = row['evento']

df_perturbacao_evento = df_perturbacao[df_perturbacao["evento"].notnull()]
df_perturbacao_evento.sample(10)

id_chamado id_origem_ocorrencia data_inicio            data_fim  \
1152    17664643                   11  2022-09-04 2022-09-13 11:23:26   
1351    17683883                   11  2022-09-10 2022-09-19 07:59:30   
1196    17658280                   11  2022-09-02 2022-09-09 08:16:00   
66      17678186                    1  2022-09-08 2022-09-09 09:01:18   
931     17683971                   11  2022-09-10 2022-09-10 20:45:38   
4157    18330160                   11  2023-02-19 2023-02-27 08:56:48   
48258   18076834                   11  2022-12-30 2022-12-30 13:12:50   
259     17685026                    1  2022-09-11 2022-09-11 16:34:49   
4343    18329965                   11  2023-02-19 2023-02-27 08:50:45   
2634    17683303                   11  2022-09-10 2022-09-10 14:11:36   

      id_bairro id_territorialidade id_logradouro  numero_logradouro  \
1152         24                   2         65896                131   
1351         18                   2         70433                  8   
1196         27                   2        107276               1518   
66           26                   2         69260                 96   
931          38                   2         14449                191   
4157          5                   1        151415                 84   
48258        33                   2         79160                254   
259          48                   3         23978                239   
4343        128                   4         91330                102   
2634        122                   4         24331               1755   

      id_unidade_organizacional                  nome_unidade_organizacional  \
1152                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
1351                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
1196                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
66                          121                             Ouvidoria GM-RIO   
931                          70  GM-RIO - Guarda Municipal do Rio de Janeiro   
4157                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
48258                        70  GM-RIO - Guarda Municipal do Rio de Janeiro   
259                          70  GM-RIO - Guarda Municipal do Rio de Janeiro   
4343                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   
2634                         70  GM-RIO - Guarda Municipal do Rio de Janeiro   

       ... tempo_prazo prazo_unidade prazo_tipo   dentro_prazo   situacao  \
1152   ...        <NA>             D          F  Fora do prazo  Encerrado   
1351   ...        <NA>             D          F  Fora do prazo  Encerrado   
1196   ...        <NA>             D          F  Fora do prazo  Encerrado   
66     ...        <NA>             D          F       No prazo  Encerrado   
931    ...        <NA>             D          F       No prazo  Encerrado   
4157   ...        <NA>             D          F  Fora do prazo  Encerrado   
48258  ...        <NA>             D          F       No prazo  Encerrado   
259    ...        <NA>             D          F       No prazo  Encerrado   
4343   ...        <NA>             D          F  Fora do prazo  Encerrado   
2634   ...        <NA>             D          F       No prazo  Encerrado   

               tipo_situacao justificativa_status reclamacoes  data_particao  \
1152            Não atendido                 None           0     2022-09-01   
1351            Não atendido                 None           0     2022-09-01   
1196            Não atendido                 None           0     2022-09-01   
66     Atendido parcialmente                 None           0     2022-09-01   
931             Não atendido                 None           0     2022-09-01   
4157            Não atendido                 None           0     2023-02-01   
48258         Não constatado                 None           0     2022-12-01   
259                 Atendido            

### Q8) Quantos chamados desse subtipo foram abertos em cada evento?

In [49]:
df_perturbacao_evento["evento"].value_counts()

evento
Rock in Rio    946
Réveillon      280
Carnaval       252
Name: count, dtype: int64

### Q9) Qual evento teve a maior média diária de chamados abertos desse subtipo?


In [ ]:
df_eventos_completos['duracao_dias'] = (
    pd.to_datetime(df_eventos_completos['data_final']) - pd.to_datetime(df_eventos_completos['data_inicial'])
).dt.days + 1
df_eventos_completos

ano data_inicial  data_final       evento  \
0         02/09 a 04/09 de 2022   2022-09-02  2022-09-04  Rock in Rio   
1         08/09 a 11/09 de 2022   2022-09-08  2022-09-11  Rock in Rio   
2  30-31/12 e 01/01 (2022-2023)   2022-12-30  2023-01-01    Réveillon   
3         18/02 a 21/02 de 2023   2023-02-18  2023-02-21     Carnaval   
4  30-31/12 e 01/01 (2023-2024)   2023-12-30  2024-01-01    Réveillon   
5            10/02 a 14/02/2024   2024-02-10  2024-02-14     Carnaval   
6                           NaN   2024-09-13  2024-09-15  Rock in Rio   
7                           NaN   2024-09-19  2024-09-22  Rock in Rio   
8                           NaN   2024-12-29  2025-01-01    Réveillon   

   taxa_ocupacao  duracao_dias  
0         0.8184             3  
1         0.9451             4  
2         0.9251             3  
3         0.9554             4  
4            NaN             3  
5            NaN             5  
6            NaN             3  
7            NaN             4  
8            NaN             4

In [53]:
df_eventos_dias = (
    df_eventos_completos[['evento', 'duracao_dias']]
    .groupby('evento', as_index=False)
    .sum()
)
df_eventos_dias

evento  duracao_dias
0     Carnaval             9
1  Rock in Rio            14
2    Réveillon            10

In [57]:
duracao_carnaval = df_eventos_dias['duracao_dias'][df_eventos_dias.evento == 'Carnaval']
duracao_reveiao = df_eventos_dias['duracao_dias'][df_eventos_dias.evento == 'Réveillon']
duracao_rir = df_eventos_dias['duracao_dias'][df_eventos_dias.evento == 'Rock in Rio']

In [61]:
contagem_por_evento = df_perturbacao_evento['evento'].value_counts()

total_carnaval = contagem_por_evento.get('Carnaval', 0)
total_rir = contagem_por_evento.get('Rock in Rio', 0)
total_reveiao = contagem_por_evento.get('Réveillon', 0)


In [65]:
total_carnaval = total_carnaval.item() if isinstance(total_carnaval, pd.Series) else total_carnaval
duracao_carnaval = duracao_carnaval.item() if isinstance(duracao_carnaval, pd.Series) else duracao_carnaval

total_rir = total_rir.item() if isinstance(total_rir, pd.Series) else total_rir
duracao_rir = duracao_rir.item() if isinstance(duracao_rir, pd.Series) else duracao_rir

total_reveiao = total_reveiao.item() if isinstance(total_reveiao, pd.Series) else total_reveiao
duracao_reveiao = duracao_reveiao.item() if isinstance(duracao_reveiao, pd.Series) else duracao_reveiao

media_carnaval = total_carnaval / duracao_carnaval
media_rir = total_rir / duracao_rir
media_reveiao = total_reveiao / duracao_reveiao

print(f"Média diária - Rock in Rio: {media_rir:.2f}")
print(f"Média diária - Carnaval: {media_carnaval:.2f}")
print(f"Média diária - Réveillon: {media_reveiao:.2f}")


Média diária - Rock in Rio: 67.57
Média diária - Carnaval: 28.00
Média diária - Réveillon: 28.00


#### O Rock in Rio possui a maior média, 67.57 chamados por dia em média

### Q10) Compare as médias diárias de chamados abertos desse subtipo durante os eventos específicos (Reveillon, Carnaval e Rock in Rio) e a média diária de chamados abertos desse subtipo considerando todo o período de 01/01/2022 até 31/12/2024.

In [69]:
# Média Diária Global no período
total_chamados = len(df_perturbacao)
inicio = pd.to_datetime('2022-01-01')
fim = pd.to_datetime('2024-12-31')
dias_periodo = (fim - inicio).days + 1
media_diaria_global = total_chamados / dias_periodo

print(f"Média diária global no período: {media_diaria_global:.2f}")

Média diária global no período: 51.81


#### Os dados mostram que eventos não estão sempre associados a aumentos nos chamados por perturbação, mas vale destacar o Rock in Rio, com uma media maior do que eventos populares tradicionais.
#### Curiosamente, o Réveillon e o Carnaval tem menos chamados por dia que a média geral. O que pode estar ligado a uma tolerância social maior ao barulho nestas datas.
#### São eventos que podem ser explorados para pensar em ações de mitigação nos outros eventos.